processing/src/test/java/io/druid/query/aggregation/hyperloglog/HyperUniquesAggregationTest.java

```diff
@@ -0,0 +1,99 @@
+/*
+* Licensed to Metamarkets Group Inc. (Metamarkets) under one
+* or more contributor license agreements. See the NOTICE file
+* distributed with this work for additional information
+* regarding copyright ownership. Metamarkets licenses this file
+* to you under the Apache License, Version 2.0 (the
+* "License"); you may not use this file except in compliance
+* with the License. You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied. See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*/
+
+package io.druid.query.aggregation.hyperloglog;
+
+import com.google.common.collect.Lists;
+import com.google.common.io.Files;
+import com.metamx.common.guava.Sequence;
+import com.metamx.common.guava.Sequences;
+import io.druid.data.input.MapBasedRow;
+import io.druid.granularity.QueryGranularity;
+import io.druid.jackson.AggregatorsModule;
+import io.druid.query.aggregation.AggregationTestHelper;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.io.File;
+
+public class HyperUniquesAggregationTest
+{
+  @Test
+  public void testIngestAndQuery() throws Exception
+  {
+    AggregationTestHelper helper = new AggregationTestHelper(Lists.newArrayList(new AggregatorsModule()));
+
+    String metricSpec = "[{"
+                        + "\"type\": \"hyperUnique\","
+                        + "\"name\": \"index_hll\","
+                        + "\"fieldName\": \"market\""
+                        + "}]";
+
+    String parseSpec = "{"
+                       + "\"type\" : \"string\","
+                       + "\"parseSpec\" : {"
+                       + "    \"format\" : \"tsv\","
+                       + "    \"timestampSpec\" : {"
+                       + "        \"column\" : \"timestamp\","
+                       + "        \"format\" : \"auto\""
+                       + "},"
+                       + "    \"dimensionsSpec\" : {"
+                       + "        \"dimensions\": [],"
+                       + "        \"dimensionExclusions\" : [],"
+                       + "        \"spatialDimensions\" : []"
+                       + "    },"
+                       + "    \"columns\": [\"timestamp\", \"market\", \"quality\", \"placement\", \"placementish\", \"index\"]"
+                       + "  }"
+                       + "}";
+
+    File f = Files.createTempDir();
+    helper.createIndex(
+        new File(this.getClass().getClassLoader().getResource("druid.sample.tsv").getFile()),
+        parseSpec,
+        metricSpec,
+        f,
```

Comment:
if createIndex(..) create tmp directory to store segment and returns it then it becomes caller's responsibility to clean it up even if caller did not really create it. That is fine too.

However I am going to create another wrapper method inside helper class that does not ask user to pass tmp file and does whole thing... something like below . let me know if that does not look Ok to you.

  public Sequence<Row> createIndexAndRunQueryOnSegment(
      File inputDataFile,
      String parserJson,
      String aggregators,
      long minTimestamp,
      QueryGranularity gran,
      int maxRowCount,
      String groupByQueryJson
  ) throws Exception
  {
..
}